scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,272 @@
+
+```{r}
+pbmc_sce_percellqc <- scater::addPerCellQC(
+  pbmc_sce_raw, 
+  subsets = list(mito = mito_genes))
+
+# we'll need this later, so pop it in now
+pbmc_sce_percellqc <- scater::addPerFeatureQC(pbmc_sce_percellqc)
+
+
+# Filter out 0s, and see before/after dims:
+dim(pbmc_sce_percellqc) 
+pbmc_sce_percellqc <- pbmc_sce_percellqc[, pbmc_sce_percellqc$total > 0]
+dim(pbmc_sce_percellqc) 
+```
+
+
+Before proceeding with further filtering, we'll want to re-arrange the SCE object to handle the RNA and ADT separately.
+Currently, the ADTs and genes are both part of the primary experiment, so let's move the ADT counts into an altExp slot.
+
+```{r}
+gene_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Gene Expression") %>% 
+  dplyr::pull(ID)
+adt_names <- as.data.frame(rowData(pbmc_sce_percellqc)) %>% 
+  dplyr::filter(Type == "Antibody Capture") %>% 
+  dplyr::pull(ID)
+
+adt_pbmc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% adt_names]
+
+# split out the experiments:
+pbmc_sce_percellqc <- pbmc_sce_percellqc[rownames(pbmc_sce_percellqc) %in% gene_names]
+altExp(pbmc_sce_percellqc, "CITEseq") <- adt_pbmc
+```
+
+
+
+### Filter RNA
+
+#### First, with `DropletUtils`
+
+```{r}
+# Filter RNA
+droplet_df <- DropletUtils::emptyDrops(
+  counts(pbmc_sce_percellqc), 
+  BPPARAM = BiocParallel::MulticoreParam(4)
+) 
+
+# Filter down to low FDR cells and check out dimensions
+# Values here are mostly NA.
+# NOTE: this filtering does propagate to altExp!! :) 
+cells_to_retain <- which(droplet_df$FDR <= 0.01)
+filtered_sce <- pbmc_sce_percellqc[, cells_to_retain]
+dim(filtered_sce) 
+```
+
+
+#### We can also filter RNA with `miQC`
+
+Mito QC stats are already present in `pbmc_sce_percellqc`.
+This is sometimes working and sometimes not for me, so commenting out for now...
+```{r}
+#miqc_model <- miQC::mixtureModel(pbmc_sce_percellqc) 
+#filtered_sce_miqc <- miQC::filterCells(pbmc_sce_percellqc,
+#                                       model = miqc_model)
+#
+#dim(filtered_sce_miqc)
+```
+Moving forward, one can choose to proceed with either `filtered_sce_miqc` or `filtered_sce`; we'll use `filtered_sce` for now.
+
+
+#### Compare `DropletUtils` to `miQC` filtering
+
+We may want to compare certain aspects of cells that were filtered out with DropletUtils vs miQC.
+Chunk below, currently pending content..., can do that!
+
+```{r}
+
+```
+
+### Filter ADT
+
+This data does have a control antibody (IgG1), so we can leverage this ADT for QC and filtering.
+```{r}
+# Filter ADT
+# http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#applying-custom-qc-filters
+
+controls <- grep("^Ig", rownames(altExp(filtered_sce)))
+adt_qc_df <- DropletUtils::cleanTagCounts(altExp(filtered_sce), controls=controls) # I feel like this control might be crap!
+
+summary(adt_qc_df$high.controls) # these are all NA????????
+summary(adt_qc_df$discard) # these are MOSTLY NA????????
```

Comment:
Definitely seems worth investigating this... I'll come back to it when I have better computational resources.